src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -13,21 +13,23 @@ class NotifyService(
   private val client: NotificationClient,
 ) {
   fun referralSent(referenceNumber: String, location: URI, recipientEmail: String) {
-    if (enabled) {
-      try {
-        client.sendEmail(
-          referralSentTemplateID,
-          recipientEmail,
-          mapOf(
-            "referenceNumber" to referenceNumber,
-            "referralUrl" to location.toString(),
-          ),
-          null,
-        )
-      } catch (e: Exception) {
-        // fixme: this failure is super important and we need a better way to reason about async errors of this nature
-        log.error("referral sent notification failed", e)
-      }
+    if (!enabled) {
+      return
+    }
+
+    try {
+      client.sendEmail(
+        referralSentTemplateID,
```

Comment:
Given we couple the configuration of this class to be a notification about a sent referral, we could also make it a listener to sent referral. Not sure if that makes things worse or better 🤷‍♂️ 
```kt
-class NotifyService(...)
+class NotifyProviderWithReferral(...) : ApplicationListener<ReferralEvent>
```